types/services.d.ts

```diff
@@ -26,17 +29,14 @@ export type AdminPaginatedResponse<T> = {
   result: Array<T>;
 } & StrapiResponseMeta;
 
-export type FindAllFlatProps = {
+export type FindAllFlatProps<T, TFields = keyof T> = {
   query: {
     threadOf?: number | string | null;
     [key: string]: any;
   } & {};
-  populate?: {
-    [key: string]: any;
-  };
-  sort?: {
-    [key: string]: any;
-  };
+  populate?: StringMap<any>;
+  sort?: StringMap<any>;
```

Comment:
```suggestion
  populate?: StringMap<unknown>;
  sort?: StringMap<unknown>;
```
Or `ToBeFixed`s. We don't want to loose those in a memory hole. Aren't we?